htlcswitch/link.go

```diff
@@ -400,6 +405,13 @@ func (l *channelLink) Start() error {
 	l.overflowQueue.Start()
 	l.hodlQueue.Start()
 
+	if l.cfg.TowerClient != nil {
+		err := l.cfg.TowerClient.RegisterChannel(l.ChanID())
```

Comment:
What's the impact of re-registering each time the link is created? Should this instead be done once for all active channels once the daemon starts up?

Reply:
the first execution will generate a sweep pkscript and cause a db write, after that it always hits an in-memory cache protected by a mutex.

i did consider that, but by placing it here we ensure we only generate pkscripts for usable channels and it is the most natural place from a perspective of lazily registering the channel and not make any assumptions about whether this is called before being added to the switch